### PR TITLE
TA 528 Additional mods for Care Leavers/Child Migrant Exemption Criteria Satisfied codes

### PR DESCRIPTION
**Pull request by solomonahmed**
## What does this pull request do?

Please describe what you did.
Changes to the feature tests for the use of the care leaver/child migrant exemption criteria satisfied code in immigration & asylum claims.  Updates were necessary after the rules governing validation were confirmed with the business.   The test cases covered in the manual submission tests cover all possible scope combinations for the matter types.  These include cases where the matter type combination is in-scope, only in-scope > 01-APR-2013 or out of scope unless a ECS code is used.  

The care leaver code should only pass validation if it is accompanied with an ECF Ref irrespective of the scoping rules.

The child migrant code should only pass validation if used with an ECF Ref and ECF Matter procurement area/access point.

New bulkload xml files were also created and old ones removed.

Extra tags added to some of the feature tests to allow fine grained control when running.

## Why make these changes?

Please describe why the changes were needed.
Updates were necessary after the rules governing validation were confirmed with the business.  Also, some of the current tests reference error messages.  These tests have been re-written to simply reference an 'error' rather than look for a specific error message.

## Checklist

- [x] Provided link to a JIRA ticket: https://dsdmoj.atlassian.net/browse/TA-528
- [x] Provided link to a JIRA ticket: https://dsdmoj.atlassian.net/browse/TA-529

### CODE REVIEW

File: features/check_care_leaver_validation_for_immigration.feature [hunk 1]
```diff
@@ -4,79 +4,74 @@ Feature: Check Care Leaver Validation for Immigration claims
...
+    | case id | mt | ecs code | ecf ref | case start date | pa | ap |
+		| 401 | IMXL:IRVL | LE001 |           | 01/11/19 | PA00140 | AP00198 |
+		| 402 | IMXL:IRVL | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+		| 403 | IMXL:IRVL | LE001 |           | 01/11/19 | PA20000 | AP20000 |
+    | 404 | IMLB:ICZN | LE001 |           | 01/11/19 | PA00140 | AP00198 |
+    | 405 | IMLB:ICZN | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+    | 406 | IMLB:ICZN | LE001 |           | 01/11/19 | PA20000 | AP20000 |
+    | 407 | IMXC:IREM | LE001 |           | 01/11/19 | PA00140 | AP00198 |
+    | 408 | IMXC:IREM | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+    | 409 | IMXC:IREM | LE001 |           | 01/11/19 | PA20000 | AP20000 |
```
Comment (your) [inline features/check_care_leaver_validation_for_immigration.feature]: ✅

File: features/check_care_leaver_validation_for_immigration.feature [hunk 2]
```diff
@@ -4,79 +4,74 @@ Feature: Check Care Leaver Validation for Immigration claims
...
-    |	401	|	IACA:IFRA	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	402	|	IACA:IFST	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	404	|	IACA:IIRC	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	405	|	IACA:ILEA	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
+    Then the outcome does not save and an error message appears
+    Examples:
+    | case id | mt | ecs code | ecf ref | case start date | pa | ap |
+		| 401 | IMXL:IRVL | LE001 |           | 01/11/19 | PA00140 | AP00198 |
+		| 402 | IMXL:IRVL | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+		| 403 | IMXL:IRVL | LE001 |           | 01/11/19 | PA20000 | AP20000 |
```
Comment (your) [inline features/check_care_leaver_validation_for_immigration.feature]: Minor: indentation issue

File: features/check_care_leaver_validation_for_immigration.feature [hunk 3]
```diff
@@ -4,79 +4,74 @@ Feature: Check Care Leaver Validation for Immigration claims
...
-    |	501	|	IACA:IFRA	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	502	|	IACA:IFST	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	504	|	IACA:IIRC	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	505	|	IACA:ILEA	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
+    Then the outcome saves successfully
+    Examples:
+    | case id | mt | ecs code | ecf ref | case start date | pa | ap |
+    | 410 | IMXL:IRVL | LE001 | 1234567AB | 01/11/19 | PA20000 | AP20000 |
+    | 411 | IMLB:ICZN | LE001 | 1234567AB | 01/11/19 | PA20000 | AP20000 |
+    | 412 | IMXC:IREM | LE001 | 1234567AB | 01/11/19 | PA20000 | AP20000 |
```
Comment (your) [inline features/check_care_leaver_validation_for_immigration.feature]: ✅

File: features/check_care_leaver_validation_for_immigration.feature [hunk 4]
```diff
@@ -4,79 +4,74 @@ Feature: Check Care Leaver Validation for Immigration claims
...
+    | case id | mt | ecs code | ecf ref | case start date | pa | ap |
+    | 413 | IACA:IFRA | LE001 |           | 01/11/19 | PA00140 | AP00198 |
+		| 414 | IACA:IFRA | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+		| 415 | IACA:IFRA | LE001 |           | 01/11/19 | PA20000 | AP20000 |
+    | 416 | IAXL:IOUT | LE001 |           | 01/11/19 | PA00140 | AP00198 |
+    | 417 | IAXL:IOUT | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+    | 418 | IAXL:IOUT | LE001 |           | 01/11/19 | PA20000 | AP20000 |
+    | 419 | IACA:IASY | LE001 |           | 01/11/19 | PA00140 | AP00198 |
+    | 420 | IACA:IASY | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+    | 421 | IACA:IASY | LE001 |           | 01/11/19 | PA20000 | AP20000 |
```
Comment (your) [inline features/check_care_leaver_validation_for_immigration.feature]: ✅

File: features/check_care_leaver_validation_for_immigration.feature [hunk 5]
```diff
@@ -4,79 +4,74 @@ Feature: Check Care Leaver Validation for Immigration claims
...
-			| 601 | IMCA:IDOM | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
-			| 602 | IMCA:IEMP | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
-			| 603 | IMCA:IEUL | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
-			| 604 | IMCA:IFME | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+    Then the outcome does not save and an error message appears
+    Examples:
+    | case id | mt | ecs code | ecf ref | case start date | pa | ap |
+    | 413 | IACA:IFRA | LE001 |           | 01/11/19 | PA00140 | AP00198 |
+		| 414 | IACA:IFRA | LE001 | 1234567AB | 01/11/19 | PA00140 | AP00198 |
+		| 415 | IACA:IFRA | LE001 |           | 01/11/19 | PA20000 | AP20000 |
```
Comment (your) [inline features/check_care_leaver_validation_for_immigration.feature]: Minor: indentation issue

File: features/check_care_leaver_validation_for_immigration.feature [hunk 6]
```diff
@@ -4,79 +4,74 @@ Feature: Check Care Leaver Validation for Immigration claims
...
-    |	701	|	IMCA:IDOM	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	702	|	IMCA:IEMP	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	703	|	IMCA:IEUL	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
-    |	704	|	IMCA:IFME	|	LE001	|	1234567AB	|	01/11/19	|	 PA00140 	|	 AP00198 	|
+    Then the outcome saves successfully
+    Examples:
+    | case id | mt | ecs code | ecf ref | case start date | pa | ap |
+    | 422 | IACA:IFRA | LE001 | 1234567AB | 01/11/19 | PA20000 | AP20000 |
+    | 423 | IAXL:IOUT | LE001 | 1234567AB | 01/11/19 | PA20000 | AP20000 |
+    | 424 | IACA:IASY | LE001 | 1234567AB | 01/11/19 | PA20000 | AP20000 |
```
Comment (your) [inline features/check_care_leaver_validation_for_immigration.feature]: ✅

File: features/check_care_leaver_validation_for_immigration.feature (hunk 2 above)
Comment (solomonahmed) [reply features/check_care_leaver_validation_for_immigration.feature]: Arghh.  Why does Atom do this!  I'll look into this.  I'm sure I resolved it.

**Pull request merged**